File: ios/Classes/FlutterQwenPlugin.m
// Stub for Flutter iOS plugin registration.
// This package uses Dart FFI and DynamicLibrary.process() on iOS;
// the host app must link the llama.cpp library (see ios/README.md).
#import <Flutter/Flutter.h>

@interface FlutterQwenPlugin : NSObject <FlutterPlugin>
@end

@implementation FlutterQwenPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  // No method channel; FFI is used from Dart.
}
@end

File: android/src/main/kotlin/dev/flutter/flutter_qwen/FlutterQwenPlugin.kt
package dev.flutter.flutter_qwen

import io.flutter.embedding.engine.plugins.FlutterPlugin

/**
 * Stub for Flutter Android plugin registration.
 * This package uses Dart FFI and loads libllama.so on Android;
 * the native library is built by this module's CMake and packed into the app.
 */
class FlutterQwenPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        // Load libllama.so so Dart's DynamicLibrary.open("libllama.so") can find it.
        System.loadLibrary("llama")
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}
